### PR TITLE
Add new CI on riscv-tests based on mill

### PR DESCRIPTION
**Pull request by ZenithalHourlyRate**
**Related issue**: follow up of #3182

**Type of change**: other enhancement

**Impact**: API addition (no impact on existing code)

**Development Phase**: implementation

**Release Notes**

### CODE REVIEW

**Comment by sequencer** [comment]
" New Continuous Integration / riscv-tests (pull_request) Successful in 63m "
two options:
1. split tests into matrix buckets to let GitHub run them in different machines.
2. run test in a powerful local CI runner

**Comment by sequencer** [comment]
please remove the duplicated cases in original CI.

**Comment by jerryz123** [comment]
Can we leave the old tests there? 
Or at most decrease the amount of stuff that runs under the old tests to just one or two configs.

**Comment by jerryz123** [comment]
Rename new-ci.yml to something more descriptive. mill-ci.yml?

**Comment by ZenithalHourlyRate** [comment]
> Rename new-ci.yml to something more descriptive. mill-ci.yml?

Done

File: build.sc [hunk 1]
```diff
@@ -220,60 +227,101 @@ object `riscv-tests` extends Module {
...
+        PathRef(T.dest)
       } else {
-        os.move(T.dest / s"$name.running.log", T.dest / s"$name.passed.log")
-        T.dest / s"$name.passed.log"
-      })
+        System.out.println(s"Running: ${emulator(top, config).elf().path} ${bin.path}")
+        val p = os.proc(emulator(top, config).elf().path, bin.path).call(stdout = T.dest / s"$name.running.log", mergeErrIntoOut = true, check = false)
+        PathRef(if (p.exitCode != 0) {
+          os.move(T.dest / s"$name.running.log", T.dest / s"$name.failed.log")
+          throw new Exception(s"Test $name failed with exit code ${p.exitCode}")
```
Comment (sequencer) [inline build.sc:318]:
```suggestion
```
Remove this and collect at last

**Review by sequencer — APPROVED** [review]

**Pull request merged**